hub/apps/get-started/uno-simple-photo-viewer.md

```diff
@@ -0,0 +1,550 @@
+
+```csharp
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using Windows.Storage;
+
+namespace SimplePhotosUno;
+
+public class ImageFileInfo : INotifyPropertyChanged
+{
+    public ImageFileInfo(StorageFile imageFile,
+        string name,
+        string type,
+        Uri uri)
+    {
+        ImageName = name;
+        ImageFileType = type;
+        ImageFile = imageFile;
+        ImageUri = uri;
+    }
+
+    public StorageFile ImageFile { get; }
+
+    public Uri ImageUri { get; }
+
+    public string ImageName { get; }
+
+    public string ImageFileType { get; }
+
+    public event PropertyChangedEventHandler? PropertyChanged;
+
+    protected void OnPropertyChanged([CallerMemberName] string? propertyName = null) =>
+        PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+}
+```
+
+The `ImageFileInfo` class now contains a `Uri` property that will be used to load the image file. Both of the methods pertaining to getting an image source or thumbnail manually are now unneeded and can be removed. We have also omitted the `ImageDimensions` property because the `ImageProperties` type is not supported on all platforms. This `ImageFileInfo` class will be used to represent the image files in the `GridView`. We will now make changes to the `MainPage.xaml` file to accommodate these changes.
+
+### Displaying images in GridView
+
+In `MainPage.xaml`, replace the GridView element with the following code:
+
+```xml
+<GridView x:Name="ImageGridView"
+        ItemsSource="{x:Bind Images, Mode=OneWay}"
+        ItemContainerStyle="{StaticResource ImageGridView_ItemContainerStyle}"
+        ItemTemplate="{StaticResource ImageGridView_ItemTemplate}"
+        ItemsPanel="{StaticResource ImageGridView_ItemsPanelTemplate}"/>
+```
+
+Finally, modify the `ImageGridView_ItemTemplate` resource to use the `ImageUri` property of the `ImageFileInfo` class. We also removed elements which use the `ImageDimensions` property:
+
+```xml
+<DataTemplate x:Key="ImageGridView_ItemTemplate"
+        x:DataType="local:ImageFileInfo">
+    <Grid Height="300"
+        Width="300"
+        Margin="8">
+        <Grid.RowDefinitions>
+            <RowDefinition />
+            <RowDefinition Height="Auto" />
+        </Grid.RowDefinitions>
+
+        <Image x:Name="ItemImage"
+            Source="{x:Bind ImageUri}"
+            Stretch="Uniform" />
+
+        <StackPanel Orientation="Vertical"
+                    Grid.Row="1">
+            <TextBlock Text="{x:Bind ImageTitle}"
+                    HorizontalAlignment="Center"
+                    Style="{StaticResource SubtitleTextBlockStyle}" />
+            <TextBlock Text="{x:Bind ImageFileType}"
+                    HorizontalAlignment="Center"
+                    Style="{StaticResource CaptionTextBlockStyle}" />
+            <RatingControl Value="{x:Bind ImageRating}" IsReadOnly="True"/>
+        </StackPanel>
+    </Grid>
+</DataTemplate>
+```
+
+### Running the app
```

Comment:
There are no instructions on how to run Mac one. But there is Linux, which is great.